### PR TITLE
get programme specific courses

### PR DESCRIPTION
**Pull request by Satvik200**

### CODE REVIEW

File: apps/programmes/views.py [hunk 1]
```diff
@@ -20,3 +22,22 @@ class ProgrammeViewSet(GenericViewSet, ListModelMixin, RetrieveModelMixin):
...
+    def list(self, request, *args, **kwargs):
+        queryset = self.get_queryset()
+        programme = get_object_or_404(Programme, uuid=self.kwargs.get('uuid'))
+        
+        response_data = {
+            'programme': ProgrammeSerializer(programme).data,
+            'courses': CourseSerializer(queryset, many=True).data
+        }
+        
+        return Response(response_data, status=status.HTTP_200_OK)
```
Comment (josemquinta15) [inline apps/programmes/views.py]: I think its simpler to add a method in the ProgrammeViewSet... For instance

```suggestion
    @action(detail=True, methods=["get"])
    def courses(self, request, *args, **kwargs):
        programme = self.get_object()
        courses = programme.courses.all()
        serializer = CourseSerializer(courses, many=True)
        return Response(serializer.data)
```

**Comment by Satvik200** [comment]
@josemquinta15 Can you please check now?

**Comment by Satvik200** [comment]
@josemquinta15 Can you check now?
I have added code in ProgramViewSet only and also removed almost all linter issues as well

File: apps/programmes/urls.py [hunk 2]
```diff
@@ -8,4 +8,5 @@
 
 urlpatterns = [
     path("", include(router.urls)),
+    path('programmes/<uuid:uuid>/courses/', ProgrammeViewSet, name='programme-courses')
```
Comment (josemquinta15) [inline apps/programmes/urls.py]: this is not necessary, it's included in the router of line 7

Comment (Satvik200) [reply apps/programmes/urls.py]: okay I will remove this

**Review by josemquinta15 — APPROVED** [review]

**Pull request merged**